src/backend/src/services/change-requests.services.ts

```diff
@@ -104,108 +114,215 @@ export default class ChangeRequestsService {
     if (foundCR.wbsElement.dateDeleted) throw new DeletedException('WBS Element', wbsPipe(foundCR.wbsElement));
 
     // verify that the user is not reviewing their own change request
-    if (reviewer.userId === foundCR.submitterId) throw new AccessDeniedException();
-
+    if (reviewer.userId === foundCR.submitterId)
+      throw new AccessDeniedException(
+        `User ${reviewer.userId} is not allowed to review their own change request submitted by User ${foundCR.submitterId}`
+      );
     // If approving a scope CR
     if (foundCR.scopeChangeRequest && accepted) {
-      // ensure a proposed solution is being approved and exists
-      if (!psId) throw new HttpException(400, 'No proposed solution selected for scope change request');
-      const foundPs = await prisma.proposed_Solution.findUnique({
-        where: { proposedSolutionId: psId }
-      });
-      if (!foundPs || foundPs.changeRequestId !== foundCR.scopeChangeRequest.scopeCrId)
-        throw new NotFoundException('Proposed Solution', psId);
-
-      // automate the changes for the proposed solution
-      // if cr is for a project: update the budget based off of the proposed solution
-      // else if cr is for a wp: update the budget and duration based off of the proposed solution
-      if (!foundCR.wbsElement.workPackage && foundCR.wbsElement.project) {
-        const newBudget = foundCR.wbsElement.project.budget + foundPs.budgetImpact;
-        const change = createChange(
-          'Budget',
-          foundCR.wbsElement.project.budget,
-          newBudget,
-          crId,
-          reviewer.userId,
-          foundCR.wbsElementId
-        );
-        await prisma.project.update({
-          where: { projectId: foundCR.wbsElement.project.projectId },
-          data: {
-            budget: newBudget
-          }
-        });
+      if (!foundCR.scopeChangeRequest.wbsProposedChanges && !psId) {
```

Comment:
you're kind of just the straw that broke the camel's back here, but this is way too much code in one method. To start off, can you make all these higher level if statements into variables above this with good names, so you can tell exactly what's happening in the if statement e.g. if(isApprovingScopeChangeRequest)

Reply:
secondly, each if statement could probably be it's own helper method. Tbh we could just make them methods on this service class as well. If ends up looking gross, feel free to move them to change request utils but I think most of these things are high level enough that they warrant a service method, just make sure they're each documented well in a comment